torch_np/_ndarray.py

```diff
@@ -449,12 +449,28 @@ def __dlpack_device__(self):
         return self.tensor.__dlpack_device__()
 
 
+def _tolist(obj):
+    """Recusrively convert tensors into lists."""
+    a1 = []
```

Comment:
Bring here the if obj is isntance list/tuple, otherwise this function assumes that obj is an iterable, whcih is a non-trivial thing to assume in this context.

Reply:
It's guarded with an if isinstance check at the call site?
https://github.com/Quansight-Labs/numpy_pytorch_interop/pull/138/files#diff-83812bd01681b687b8fced4eac464b076aecffc471e0364d905b38fba8794e6cR487

Reply:
Yeah. What I'm saying is that the logic of this function assumes that if implicitly. The if instance should be inside this function, not in the caller site..